src/rebar_hooks.erl

```diff
@@ -32,7 +32,11 @@ run_provider_hooks(Dir, Type, Command, Providers, Opts, State) ->
             State;
         AllHooks ->
             TypeHooks = proplists:get_value(Type, AllHooks, []),
-            run_provider_hooks_(Dir, Type, Command, Providers, TypeHooks, rebar_state:opts(State, Opts))
+            State1 = rebar_state:opts(State, Opts),
+            Env = create_env(State, Opts),
+            State2 = rebar_state:env(State1, Env),
+            run_provider_hooks_(Dir, Type, Command, Providers,
+                                TypeHooks, State2)
```

Comment:
I'm not sure what this provides here. So we create the ENV and store it in the provider? The bug as illustrated at https://github.com/blt/port_compiler/issues/30 seems to be that https://github.com/blt/port_compiler/pull/38/files#diff-190510d6d34e2aff57877104e5ec0246R152 uses the ENV values to just expand the ENV themselves. 

Is there a reason why we couldn't just expose `create_env` as a function instead (maybe stick it in a `rebar_env` module or something like that) and let the plugin call it and expand it however it wants based on the current opts and config? It sounds like it would be far less risky than having to track and update the `env` value in the state record along with everything else if we make it so whoever needs it generates it on-demand.